File: pattern/1.PatternSlidingWindow/c1.js
function findSumSequence(arr, k) {
  let sum = 0,
    max = 0,
    start = 0;

  for (let end = 0; end < arr.length; end++) {
    sum += arr[end];

    if (end >= k - 1) {
      max = Math.max(max, sum);
      sum -= arr[start];
      start++;
    }
  }

  return max;
}

console.log(findSumSequence([2, 1, 5, 1, 3, 2], 3));
console.log(findSumSequence([2, 3, 4, 1, 5], 2));

File: pattern/1.PatternSlidingWindow/c1.py
# Maximum Sum Subarray of Size K (easy)

def find_max_sub_array(numbers, k):
    sum = 0
    start = 0
    valMax = numbers[0]

    for idx,num in enumerate(numbers):
        sum += num
        if idx >= k - 1:
            valMax = max(valMax, sum)
            sum -= numbers[start]
            start += 1
    return valMax


arr = [2, 1, 5, 1, 3, 2]
k = 3

print(find_max_sub_array(arr, k))